azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/DeploymentApiLiveTest.java

```diff
@@ -0,0 +1,127 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.features;
+
+import org.jclouds.azurecompute.arm.domain.Deployment;
+import org.jclouds.azurecompute.arm.domain.Deployment.ProvisioningState;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiLiveTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.List;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertNotNull;
+
+@Test(groups = "live", testName = "DeploymentApiLiveTest", singleThreaded = true)
+public class DeploymentApiLiveTest extends BaseAzureComputeApiLiveTest {
+    private int maxTestDuration = 90;
+    private int pollingInterval = 3; // how frequently to poll for create status
+
+    private String resourceGroup;
+    private String resourceName;
+    private String deploymentName;
+
+    @BeforeClass
+    @Override
+    public void setup() {
+        super.setup();
+        resourceGroup = getResourceGroupName();
+    }
+
+    private String getPutBody(String template, String mode, String parameters) {
+        String body = "{ " +
+                      "\"properties\" : " +
+                      "  { " +
+                      "    \"template\" : " + template + ", " +
+                      "    \"mode\" : \"" + mode + "\", " +
+                      "    \"parameters\" : " + parameters + " " +
+                      "  } " +
+                      "}";
+        return body;
+    }
+
+    @Test(groups = "live")
+    public void testCreate() {
+        Long now = System.currentTimeMillis();
+        resourceName = "jcres" + now;
+        deploymentName = "jcdep" + now;
+        String template = "{\"$schema\":\"https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#\",\"contentVersion\":\"1.0.0.0\",\"parameters\":{\"newStorageAccountName\":{\"type\":\"string\",\"metadata\":{\"description\":\"Name of the Storage Account\"}},\"storageAccountType\":{\"type\":\"string\",\"defaultValue\":\"Standard_LRS\",\"allowedValues\":[\"Standard_LRS\",\"Standard_GRS\",\"Standard_ZRS\"],\"metadata\":{\"description\":\"Storage Account type\"}},\"location\":{\"type\":\"string\",\"allowedValues\":[\"East US\",\"West US\",\"West Europe\",\"East Asia\",\"Southeast Asia\"],\"metadata\":{\"description\":\"Location of storage account\"}}},\"resources\":[{\"type\":\"Microsoft.Storage/storageAccounts\",\"name\":\"[parameters('newStorageAccountName')]\",\"apiVersion\":\"2015-05-01-preview\",\"location\":\"[parameters('location')]\",\"properties\":{\"accountType\":\"[parameters('storageAccountType')]\"}}]}";
+        String parameters = "{\"newStorageAccountName\":{\"value\":\"" + resourceName + "\"},\"storageAccountType\":{\"value\":\"Standard_LRS\"},\"location\":{\"value\":\"West US\"}}";
+        String properties = getPutBody(template, "Incremental", parameters);
```

Comment:
I happen to disagree, because the actual API is being reflected.  There may be cases where an ARM template that isn't exposed via a builder -- maybe Azure Container Service, for instance -- may want to be used.  The user should have access to the raw deployment API do do that.

I agree builders are a better abstraction and those, to me, should be at a higher level and not necessarily imposed on the actual API implementation itself.

My biggest two concerns are the imposing of "object oriented" methods on top of a defined API that is basically pre-structured JSON (yes, I know it is error prone); and second, requiring builders reduces the generality of the API, if one were to want to use it in a manner that JClouds (and the arm provider) didn't expose directly.